sdk/src/test/java/com/qonversion/android/sdk/internal/userProperties/UserPropertiesStorageImplTest.kt

```diff
@@ -0,0 +1,451 @@
+            val existingProperties = mutableMapOf<String, String>()
+
+            mockGetPropertiesFromStorage(existingProperties)
+
+            // when
+            spykStorage.set(mapOf())
+
+            // then
+            verify(exactly = 0) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEqualTo(existingProperties)
+        }
+
+        @Test
+        fun `set properties when properties val is empty`() {
+            // given
+            val properties = mapOf("key1" to "value1", "key2" to "value2")
+            val existingProperties = mutableMapOf<String, String>()
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            mockGetPropertiesFromStorage(existingProperties)
+
+            // when
+            spykStorage.set(properties)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEqualTo(properties)
+        }
+
+        @Test
+        fun `set property when properties val is not empty`() {
+            // given
+            val properties = mapOf("key1" to "value1", "key2" to "value2")
+            val existingProperties = mutableMapOf("oldKey" to "oldValue")
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            mockGetPropertiesFromStorage(existingProperties)
+
+            // when
+            spykStorage.set(properties)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEqualTo(properties + existingProperties)
+        }
+    }
+
+    @Nested
+    inner class Delete {
+        private lateinit var spykStorage: UserPropertiesStorageImpl
+
+        @BeforeEach
+        fun setUp() {
+            spykStorage = spyk(userPropertiesStorage)
+        }
+
+        @Test
+        fun `delete non-existent property`() {
+            // given
+            val key = "keyToDelete"
+
+            val existingProperties = mutableMapOf("key" to "value")
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            every {
+                spykStorage.properties
+            } returns existingProperties
+
+            // when
+            spykStorage.delete(key)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEqualTo(existingProperties)
+        }
+
+        @Test
+        fun `delete existent property`() {
```

Comment:
I'm not sure but think this is the right way:
```suggestion
        fun `delete existing property`() {
```

And in other test names too